### ISSUE TITLE
FFmpegFrameRecorder using H264 (libx264)

### ISSUE THREAD

**Issue by duongxinh2003** [bytedeco/javacv#1312]
I want to use libx264 encoder without H264 native. It looks like   Stream #0:0 -> #0:0 (h264 (native) -> h264 (libx264)).
I used this line             recorder.setVideoCodec(avcodec.AV_CODEC_ID_H264);
            recorder.setVideoCodecName("libx264");
but it doesn't effect

**Comment by saudet** [comment]
Yes, that uses libx264 by default. What are you trying to do exactly?

**Comment by duongxinh2003** [comment]
@saudet If I use java native code, the output mp4 file cannot play in Internet Explorer, so I must add a FFmpeg command line to convert it: `ffmpeg -i abc.mp4 output.mp4`. It work well on IE, and here is put out `libavutil      56. 32.100 / 56. 32.100
  libavcodec     58. 55.100 / 58. 55.100
  libavformat    58. 30.100 / 58. 30.100
  libavdevice    58.  9.100 / 58.  9.100
  libavfilter     7. 58.100 /  7. 58.100
  libswscale      5.  6.100 /  5.  6.100
  libswresample   3.  6.100 /  3.  6.100
  libpostproc    55.  6.100 / 55.  6.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'abc.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf58.20.100
  Duration: 00:00:07.20, start: 0.000000, bitrate: 20519 kb/s
    Stream #0:0(und): Video: h264 (High 4:4:4 Predictive) (avc1 / 0x31637661), yuv420p, 640x480, 20517 kb/s, 20 fps, 20 tbr, 10240 tbn, 40 tbc (default)
    Metadata:
      handler_name    : VideoHandler
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> h264 (libx264))
Press [q] to stop, [?] for help
[libx264 @ 0000013a3982a280] using cpu capabilities: MMX2 SSE2Fast LZCNT SSSE3 SSE4.2
[libx264 @ 0000013a3982a280] profile Progressive High, level 3.0, 4:2:0, 8-bit
[libx264 @ 0000013a3982a280] 264 - core 157 r2970 5493be8 - H.264/MPEG-4 AVC codec - Copyleft 2003-2019 - http://www.videolan.org/x264.html - options: cabac=1 ref=3 deblock=1:0:0 analyse=0x3:0x113 me=hex subme=7 psy=1 psy_rd=1.00:0.00 mixed_ref=1 me_range=16 chroma_me=1 trellis=1 8x8dct=1 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=-2 threads=6 lookahead_threads=1 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=3 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=2 keyint=250 keyint_min=20 scenecut=40 intra_refresh=0 rc_lookahead=40 rc=crf mbtree=1 crf=23.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 ip_ratio=1.40 aq=1:1.00
Output #0, mp4, to 'output.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf58.30.100
    Stream #0:0(und): Video: h264 (libx264) (avc1 / 0x31637661), yuv420p, 640x480, q=-1--1, 20 fps, 10240 tbn, 20 tbc (default)
    Metadata:
      handler_name    : VideoHandler
      encoder         : Lavc58.55.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 0/0/0 buffer size: 0 vbv_delay: -1
frame=  144 fps= 65 q=-1.0 Lsize=    1060kB time=00:00:07.05 bitrate=1231.3kbits/s speed= 3.2x
video:1057kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.238244%
[libx264 @ 0000013a3982a280] frame I:1     Avg QP:23.54  size: 28938
[libx264 @ 0000013a3982a280] frame P:38    Avg QP:23.79  size: 16384
[libx264 @ 0000013a3982a280] frame B:105   Avg QP:26.37  size:  4098
[libx264 @ 0000013a3982a280] consecutive B-frames:  1.4%  4.2%  0.0% 94.4%
[libx264 @ 0000013a3982a280] mb I  I16..4:  7.1% 90.7%  2.2%
[libx264 @ 0000013a3982a280] mb P  I16..4:  1.7% 16.0%  0.2%  P16..4: 36.8% 23.0% 16.6%  0.0%  0.0%    skip: 5.7%
[libx264 @ 0000013a3982a280] mb B  I16..4:  0.4%  2.6%  0.0%  B16..8: 31.9%  7.5%  1.8%  direct: 8.3%  skip:47.5%  L0:48.3% L1:33.5% BI:18.2%
[libx264 @ 0000013a3982a280] 8x8 transform intra:88.1% inter:78.8%
[libx264 @ 0000013a3982a280] coded y,uvDC,uvAC intra: 70.0% 75.3% 20.7% inter: 25.7% 30.0% 2.2%
[libx264 @ 0000013a3982a280] i16 v,h,dc,p: 15% 20%  9% 56%
[libx264 @ 0000013a3982a280] i8 v,h,dc,ddl,ddr,vr,hd,vl,hu: 18% 15% 44%  3%  4%  4%  4%  3%  4%
[libx264 @ 0000013a3982a280] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 23% 12% 16%  6% 13% 10%  8%  7%  5%
[libx264 @ 0000013a3982a280] i8c dc,h,v,p: 57% 18% 21%  3%
[libx264 @ 0000013a3982a280] Weighted P-Frames: Y:5.3% UV:0.0%
[libx264 @ 0000013a3982a280] ref P L0: 32.7%  8.6% 41.3% 16.9%  0.5%
[libx264 @ 0000013a3982a280] ref B L0: 63.2% 28.7%  8.1%
[libx264 @ 0000013a3982a280] ref B L1: 91.1%  8.9%
[libx264 @ 0000013a3982a280] kb/s:1202.02`

**Comment by saudet** [comment]
You'll need to make you're not enabling any features that IE doesn't support. It looks like the `ffmpeg` program does that by default, so compare the log output from the `ffmpeg` program and the one from JavaCV. That will tell you which options you need to deactivate.

**Comment by duongxinh2003** [comment]
@saudet I don't understand why record using FrameRecorder can play in Chrome but IE cannot. Who can help me? Thank you.

**Comment by mifritscher** [comment]
Because IE is a beast...

Seriously, the generated video file use features which are ok for chrome, but not for IE. Try to use mediainfo on both files, perhaps you get the differences between them.
